portal/src/main/scala/cromwell/pipeline/controller/ProjectController.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.pipeline.controller
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.Directives._
+import cromwell.pipeline.datastorage.dto.UserId
+import cromwell.pipeline.datastorage.dto.project.ProjectAdditionRequest
+import cromwell.pipeline.service.{ ProjectAccessDeniedException, ProjectNotFoundException, ProjectService }
+import cromwell.pipeline.utils.auth.AccessTokenContent
+import de.heikoseeberger.akkahttpplayjson.PlayJsonSupport._
+
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+
+class ProjectController(projectService: ProjectService)(
+  implicit executionContext: ExecutionContext
+) {
+  val route: AccessTokenContent => Route = accessToken =>
+    path("projects") {
+      concat(
+        get {
+          parameter('name.as[String]) {
+            name =>
+              onComplete(projectService.getProjectByName(name, UserId(accessToken.userId))) {
+                case Success(project) => complete(project)
+                case Failure(e) =>
+                  e match {
+                    case ProjectNotFoundException("Project not found") => complete(StatusCodes.NotFound, e.getMessage)
+                    case ProjectAccessDeniedException("Access denied") => complete(StatusCodes.Forbidden, e.getMessage)
+                  }
+                case _ => complete(StatusCodes.InternalServerError)
```

Comment:
There is no need to check that your exceptions contain specific error code like `"Project not found"`. Also, you can do it with only one pattern matching as follows
```
case Success(project) =>
case Failure(e: ProjectNotFoundException) =>
case Failure(e: ProjectAccessDeniedException) =>
case Failure(e) =>
```
Notice that in the last `case` I used a named variable instead of the underscore. This is because it is better to use a message from this exception to provide an explanation of what caused the internal server error.